bionic/task_state.py

```diff
@@ -317,3 +329,54 @@ def _check_accessors_for_version_problems(self):
 
         for accessor in accessors_needing_saving:
             accessor.update_provenance()
+
+    # NOTE: This can be optimized further.
+    def new_state_for_completion(self, new_task_states_by_key, to_be_computed=True):
+        """
+        Returns a copy of task state after keeping only the necessary data
+        required for completion. Along with keeping only necessary data for
```

Comment:
phrasing: You don't need to say "keeping only necessary data for completion" twice; you can just start the first sentence with "In addition,".